src/App/App.js

```diff
@@ -44,60 +47,72 @@ export class App extends Component {
     })
   }
 
-  toggleView = (event) => {
-    event.preventDefault()
-    if (this.state.isSingleView) {
-      this.setState((prevState) => {
-        return { isSingleView: !prevState.isSingleView }
-      })
-    } else if (!this.state.isSingleView && event.target.name !== 'home') {
-      this.setState((prevState) => {
-        this.fetchData(`movies/${event.target.id}`)
-          .then((data) => {
-            this.setState({
-              singleMovie: data.movie,
-              isLoading: false,
-              isSingleView: !prevState.isSingleView
-            })
-          })
-          .catch((error) => this.setState({ isError: true }))
-      })
-    }
+  toggleSingleView = () => {
+    this.setState((prevState) => {
+      return { isSingleView: !prevState.isSingleView }
+    })
   }
 
   render() {
     return (
-      <div>
-        <Header
-          movies={this.state.movies}
-          isSingleView={this.state.isSingleView}
-          toggleView={this.toggleView}
-          filterMovies={this.filterMovies}
+      <Switch>
+        <Route
+          exact
+          path='/'
+          render={() => {
+            return (
+              <div>
+                <Header
+                  movies={this.state.movies}
+                  isSingleView={this.state.isSingleView}
+                  filterMovies={this.filterMovies}
+                />
+                {this.state.isError && (
+                  <h1 style={{ textAlign: 'center', marginTop: '5vh' }}>
+                    Server Error, try Rotten Tomatoes instead
+                  </h1>
+                )}
+                {this.state.isLoading && !this.state.isError && (
+                  <h1 style={{ textAlign: 'center', marginTop: '5vh' }}>
+                    Loading... Grab some popcorn!
+                  </h1>
+                )}
+                {!this.state.isLoading && !this.state.isError && (
+                  <Main
+                    movies={this.state.filteredMovies}
+                    isSingleView={this.state.isSingleView}
+                    toggleView={this.toggleView}
+                  />
+                )}
+              </div>
+            )
+          }}
+        />
+        <Route
+          exact
+          path='/:id'
+          render={({ match }) => {
+            return (
+              <div>
+                <Header
```

Comment:
Since the Header component is displayed for both routes, could you extract it out of the routes so that you write it only once?

Reply:
Pending answer in Slack.

Reply:
Ok, now I see you're using the `location={location.pathname}` in the Header component. If you need that data in the component, then I'm not sure how you would extract it outside of a Route and still get the `location` object.

Possibly, you could have a duplicate Route of `<Route path='/'` _without_ exact that would hold onto only the Header component.